feat(route): 增加前瞻网

moke8 opened the pull request:
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/qianzhan/analyst/column
/qianzhan/analyst/rank
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

moke8 committed:
增加前瞻网

moke8 committed:
Merge branch 'master' of https://github.com/moke8/RSSHub

github-actions[bot] committed:
style: auto format

moke8 committed:
Merge branch 'DIYgod:master' into master

TonyRL commented on lib/v2/qianzhan-analyst/column.js:
```diff
@@ -0,0 +1,55 @@
+        rootUrl = rootUrl + 'list/' + ctx.params.type + '.html';
+    }
+
+    const response = await got({
+        method: 'get',
+        url: rootUrl,
+    });
```
got request can be simplify to
```suggestion
    const response = await got(rootUrl);
```

TonyRL commented on lib/v2/qianzhan-analyst/column.js:
```diff
@@ -0,0 +1,55 @@
+            (_, item) =>
+                new Promise((resolve) => {
+                    ctx.cache.tryGet(item, async () => {
+                        const detailResponse = await got({
+                            method: 'get',
+                            url: item,
+                        });
```
got request can be simplify to
```suggestion
                        const detailResponse = await got(item);
```

TonyRL commented on lib/v2/qianzhan-analyst/column.js:
```diff
@@ -0,0 +1,55 @@
+                        const $ = cheerio.load(detailResponse.data);
+                        const description = $('#div_conbody').html();
+                        const title = $('#h_title').text();
+                        const pubDate = $('#pubtime_baidu').text().split('• ')[1];
```
```suggestion
                        const pubDate = timezone(parseDate($('#pubtime_baidu').text().split('• ')[1], 'YYYY-MM-DD HH:mm:ss'), +8);
```
docs: https://docs.rsshub.app/joinus/pub-date.html

TonyRL commented on lib/v2/qianzhan-analyst/rank.js:
```diff
@@ -0,0 +1,48 @@
+    const type = ctx.params.type === 'week' ? 1 : 2;
+    const rootUrl = 'https://www.qianzhan.com/analyst/';
+
+    const response = await got({
+        method: 'get',
+        url: rootUrl,
+    });
```
```suggestion
    const response = await got(rootUrl);
```

TonyRL commented on lib/v2/qianzhan-analyst/rank.js:
```diff
@@ -0,0 +1,48 @@
+    });
+    const $ = cheerio.load(response.data);
+    const links = $('#div_hotlist ul[idx=' + type + '] a')
+        .slice(0, 9)
```
```suggestion
```
no need to slice

TonyRL commented on lib/v2/qianzhan-analyst/rank.js:
```diff
@@ -0,0 +1,48 @@
+            (_, item) =>
+                new Promise((resolve) => {
+                    ctx.cache.tryGet(item, async () => {
+                        const detailResponse = await got({
+                            method: 'get',
+                            url: item,
+                        });
```
```suggestion
                        const detailResponse = await got(item);
```

TonyRL commented on lib/v2/qianzhan-analyst/column.js:
```diff
@@ -0,0 +1,55 @@
+                            url: item,
+                        });
+                        const $ = cheerio.load(detailResponse.data);
+                        const description = $('#div_conbody').html();
```
```suggestion
                        const description = $('#divArtBody').html();
```
`#divArtBody` seems a better choice as it does not includes the title again and those self-promoted stuff

TonyRL commented on lib/v2/qianzhan-analyst/rank.js:
```diff
@@ -0,0 +1,48 @@
+                            url: item,
+                        });
+                        const $ = cheerio.load(detailResponse.data);
+                        const description = $('#div_conbody').html();
```
```suggestion
                        const description = $('#divArtBody').html();
```

TonyRL commented on lib/v2/qianzhan-analyst/rank.js:
```diff
@@ -0,0 +1,48 @@
+                        const $ = cheerio.load(detailResponse.data);
+                        const description = $('#div_conbody').html();
+                        const title = $('#h_title').text();
+                        const pubDate = $('#pubtime_baidu').text().split('• ')[1];
```
```suggestion
                        const pubDate = timezone(parseDate($('#pubtime_baidu').text().split('• ')[1], 'YYYY-MM-DD HH:mm:ss'), +8);
```

TonyRL commented on lib/v2/qianzhan-analyst/router.js:
```diff
@@ -0,0 +1,4 @@
+module.exports = (router) => {
+    router.get('/rank/:type?', require('./rank'));
+    router.get('/column/:type?', require('./column'));
```
Please sort them in alphabetical order

TonyRL commented on docs/finance.md:
```diff
@@ -250,6 +250,28 @@ TokenInsight 官方亦有提供 RSS，可参考 <https://api.tokeninsight.com/re
+
+### 文章列表
+
+<Route author="moke8" example="/qianzhan-analyst/column/all" path="/qianzhan-analyst/column/:type" :paramsDesc="['分类，见下表']">
```
The path here should follow `router.js`
```suggestion
<Route author="moke8" example="/qianzhan-analyst/column/all" path="/qianzhan-analyst/column/:type?" :paramsDesc="['分类，见下表']">
```

TonyRL commented on docs/finance.md:
```diff
@@ -250,6 +250,28 @@ TokenInsight 官方亦有提供 RSS，可参考 <https://api.tokeninsight.com/re
+
+### 排行榜
+
+<Route author="moke8" example="/qianzhan-analyst/rank/week" path="/qianzhan-analyst/rank/:type" :paramsDesc="['分类，见下表']">
```
The path here should follow `router.js`
```suggestion
<Route author="moke8" example="/qianzhan-analyst/rank/week" path="/qianzhan-analyst/rank/:type?" :paramsDesc="['分类，见下表']">
```

TonyRL reviewed (commented):
Missing `maintainer.js`
Using path like `/qianzhan/analyst/column/:type?` and `/qianzhan/analyst/rank/:type?` seems a better choice as the domain name is `qianzhan.com` instead of `qianzhan-analyst.com`

dependabot[bot] committed:
chore(deps): bump rand-user-agent from 1.0.77 to 1.0.79 (#137)

Bumps [rand-user-agent](https://github.com/WebScrapingAPI/rand-user-agent) from 1.0.77 to 1.0.79.
- [Release notes](https://github.com/WebScrapingAPI/rand-user-agent/releases)
- [Commits](https://github.com/WebScrapingAPI/rand-user-agent/commits)

---
updated-dependencies:
- dependency-name: rand-user-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

TonyRL committed:
Merge branch 'DIYgod/master' into pr/moke8/10524

TonyRL committed:
fix: apply suggestions from cr

TonyRL committed:
fix: deepscan warning
